pkg/sql/plan/build_ddl.go

```diff
@@ -297,8 +297,10 @@ func buildTableDefs(defs tree.TableDefs, ctx CompilerContext, tableDef *TableDef
 
 				if _, ok := attr.(*tree.AttributeAutoIncrement); ok {
 					auto_incr = true
-					if colType.GetId() != int32(types.T_int32) && colType.GetId() != int32(types.T_int64) {
-						return errors.New(errno.InvalidColumnDefinition, "the auto_incr column is only support int32 and int64 type now")
+					id := types.T(colType.GetId())
+					if !(id == types.T_int8 || id == types.T_int16 || id == types.T_int32 || id == types.T_int64 ||
```

Comment:
Check if types has an IsInt or IsUint method.  If not, add one.

Reply:
ok